lib/foreman_maintain/package_manager/yum.rb

```diff
@@ -0,0 +1,129 @@
+module ForemanMaintain::PackageManager
+  class Yum < Base
+    VERSIONLOCK_START_CLAUSE = '## foreman-maintain - start'.freeze
+    VERSIONLOCK_END_CLAUSE = '## foreman-maintain - end'.freeze
+    VERSIONLOCK_CONFIG_FILE = '/etc/yum/pluginconf.d/versionlock.conf'.freeze
+    VERSIONLOCK_DEFAULT_LIST_FILE = '/etc/yum/pluginconf.d/versionlock.list'.freeze
+
+    def foreman_related_packages
+      installed = sys.execute("repoquery -a --qf='%{envra} %{repo.baseurl}'").
+                  split("\n"). map(&:split)
+      repos_to_lock = []
+      foreman_repo = installed.find { |package| package[0] =~ /foreman-installer-\d+/ }[1]
+      repos_to_lock << foreman_repo
+      installed.inject([]) do |related, package|
```

Comment:
I think you can also save this iteration, something like below,

Query repoid,
~~~
repoquery -a --qf='%{envra} %{repo.id}' |grep foreman-installer
1:foreman-installer-1.18.0.3-1.el7sat.noarch rhel-7-server-satellite-6.4-rpms
0:foreman-installer-katello-3.7.0.15-1.el7sat.noarch rhel-7-server-satellite-6.4-rpms
~~~

and then directly query all packages which are there from that repoid,

~~~
repoquery -a --repoid=rhel-7-server-satellite-6.4-rpms
~~~